cmds/fwtools/spidev/spidev_linux.go

```diff
@@ -77,8 +77,10 @@ func run(args []string, spiOpen spiOpenFunc, input io.Reader, output io.Writer)
 		return err
 	}
 	defer s.Close()
-	if err := s.SetSpeedHz(*speed); err != nil {
-		return err
+	if *speed != 0 {
```

Comment:
The last thing that we do in spiOpen is setting speed to 500000 Hz and then if *speed is not zero we reset speed again? That looks strange (to set the speed we need two steps), but If this is only way how it works because of $REASONS maybe some comment will be nice.

Reply:
I added a note, see if that is enough.